common/analysis/lint_waiver.cc

```diff
@@ -180,4 +211,237 @@ void LintWaiverBuilder::ProcessTokenRangesByLine(
+
+        can_use_lineno = true;
+        continue;
+      }
+
+      if (arg == "regex") {
+        // Pre-compile regex to see if it's valid
+        regex = val;
+        try {
+            std::regex(std::string(regex));
+        } catch (std::regex_error& e) {
+          WaiveCommandError(token_pos, filename,
+                            absl::StrCat("Invalid regex: ", e.what()));
+          return false;
+        }
+        can_use_regex = true;
+        continue;
+      }
+
+      WaiveCommandError(token_pos, filename,
+                        absl::StrCat("Unsupported flag: ", arg));
+      return false;
+
+    case CFG_TK_NEWLINE:
+      // Check if everything required has been set
+      if (rule == nullptr || (!can_use_regex && !can_use_lineno)) {
+        WaiveCommandError(token_pos, filename,
+                          "Insufficient waiver configuration");
+        return false;
+      }
+
+      if (can_use_regex && can_use_lineno) {
+        WaiveCommandError(token_pos, filename,
+                          "Regex and line flags are mutually exclusive");
+        return false;
+      }
+
+      if (can_use_regex) {
+        waiver->WaiveWithRegex(rule, regex);
+      }
+
+      if (can_use_lineno) {
+        waiver->WaiveLineRange(rule, line_start - 1, line_end);
+      }
+
+      return true;
+    default:
+      WaiveCommandError(token_pos, filename, "Expecting arguments");
+      return false;
+    }
+  }
+
+  return true;
+}
+
+static const std::map<absl::string_view,
+                     std::function<bool(const TokenRange& r,
+                                        absl::string_view,
+                                        absl::string_view,
+                                        const LineColumnMap&,
+                                        LintWaiver*,
+                                        const std::set<absl::string_view>
+                                            &active_rules
+                                       )>> &GetCommandHandlers() {
+  // allocated once, never freed
+  static const auto* handlers = new std::map<absl::string_view,
+                  std::function<bool(const TokenRange& r,
+                                     absl::string_view,
+                                     absl::string_view,
+                                     const LineColumnMap&,
+                                     LintWaiver*,
+                                     const std::set<absl::string_view>
+                                         &active_rules
+                                    )>> {
+    {"waive", WaiveCommandHandler},
+  };
+  return *handlers;
+}
+
+bool LintWaiverBuilder::ApplyExternalWaivers(
+    const std::set<absl::string_view> &active_rules,
+    absl::string_view filename,
+    absl::string_view waivers_config_content) {
+
+  if (waivers_config_content == nullptr) {
+    return false;
+  }
+
+  ConfigFileLexer lexer{waivers_config_content};
+  const LineColumnMap line_map(waivers_config_content);
```

Comment:
Did you want to pass in `const LineColumnMap& line_map` into this function?

Reply:
That's what I was asking about, but it I misunderstood what you meant in the other comment